shared/teams/team/nav-header/index.new.tsx

```diff
@@ -0,0 +1,266 @@
+//       small={true}
+//       type="Default"
+//       mode="Secondary"
+//     />
+//     <AddPeopleHow
+//       attachTo={props.getAttachmentRef}
+//       onHidden={props.toggleShowingMenu}
+//       teamID={props.teamID}
+//       visible={props.showingMenu}
+//     />
+//   </>
+// )
+// const AddPeopleButton = Kb.OverlayParentHOC(_AddPeopleButton)
+
+type HeaderTitleProps = {
+  active: boolean
+  canAddPeople: boolean
+  canChat: boolean
+  canEdit: boolean
+  description: string
+  isOpen: boolean
+  loading: boolean
+  location?: string
+  members: number
+  newMemberCount?: string
+  onAddSelf?: () => void
+  onChat: () => void
+  onEdit: () => void
+  onEditAvatar?: () => void
+  onEditDescription?: () => void
+  onRename?: () => void
+  onShare: () => void
+  role: string
+  teamID: TeamID
+  teamname: string
+} & Kb.OverlayParentProps
+
+const roleDisplay = {
+  admin: 'an admin of',
+  none: 'not a member of',
+  owner: 'an owner of',
+  reader: 'a reader in',
+  writer: 'a writer in',
+}
+
+const _HeaderTitle = (props: HeaderTitleProps) => {
+  const avatar = (
+    <Kb.Avatar
+      editable={!!props.onEditAvatar}
+      onEditAvatarClick={props.onEditAvatar}
+      teamname={props.teamname}
+      size={96}
+      style={Styles.collapseStyles([
+        styles.alignSelfFlexStart,
+        props.onEditAvatar && styles.marginBottomRightTiny, // space for edit icon
+        props.onEditAvatar && styles.clickable,
+      ])}
+    />
+  )
+
+  const topDescriptors = (
+    <Kb.Box2 direction="vertical" alignSelf="flex-start" gap="xtiny">
+      <Kb.Box2 direction={Styles.isMobile ? 'vertical' : 'horizontal'} gap="xtiny" alignSelf="flex-start">
+        <Kb.Box2 direction="horizontal" alignItems="flex-end" gap="xtiny" alignSelf="flex-start">
+          <Kb.Text type="Header" lineClamp={1}>
+            {props.teamname}
+          </Kb.Text>
+          {!!props.onRename && <Kb.Icon type="iconfont-edit" onClick={props.onRename} />}
+        </Kb.Box2>
+        {props.isOpen && (
+          <Kb.Meta title="open" backgroundColor={Styles.globalColors.green} style={styles.openMeta} />
+        )}
+      </Kb.Box2>
+      {!!props.role && (
+        <Kb.Box2 direction="horizontal" gap="xtiny" alignSelf="flex-start">
+          {(props.role === 'admin' || props.role === 'owner') && (
+            <Kb.Icon
+              color={props.role === 'owner' ? Styles.globalColors.yellowDark : Styles.globalColors.black_35}
+              fontSize={Styles.isMobile ? 16 : 10}
+              type={props.role === 'owner' ? 'iconfont-crown-owner' : 'iconfont-crown-admin'}
+            />
+          )}
+          {(!Styles.isMobile || !!props.role) && (
+            <>
+              <Kb.Text type="BodySmall">
+                {Styles.isMobile
+                  ? capitalize(props.role)
+                  : `You are ${roleDisplay[props.role] || 'a member of'} this team. `}
+              </Kb.Text>
+              {props.role === 'none' && (
```

Comment:
<img width="438" alt="image" src="https://user-images.githubusercontent.com/11968340/73696490-628cca00-46aa-11ea-9309-36d1994605a6.png">
Team name needs ellipsis; it isn't clear that "Add yourself" is a link

Reply:
I'd rather make the team name multiple lines than ellipsize it. We don't have a blessed style for underlining that link; should I just manually style it?

Reply:
Multiline name sgtm. Not sure, I wonder if the blue "add yourself" banner will exist on mobile too. Fine to leave as is for now.